tracer/src/Datadog.Trace/Tagging/TraceTagCollection.cs

```diff
@@ -81,6 +104,17 @@ private bool SetTag(string name, string? value, bool replaceIfExists)
                 return RemoveTag(name);
             }
 
+            if (name == Trace.Tags.Propagated.DecisionMaker)
```

Comment:
I assume value can never be null in that case, but have you checked? because if it's possible you do not cover that case anymore (and maybe we should cover it anyway for the sake of future proofing)

Reply:
Value cannot be null, in fact if the value is null we call `RemoveTag` above. So the tag must have a value in order to exist.

We also only set this explicitly when the mechanism (value) is not null: https://github.com/DataDog/dd-trace-dotnet/blob/f536010c83a7b4c1ebe207fa090dd0aa72237f6d/tracer/src/Datadog.Trace/TraceContext.cs#L207